sender.go

```diff
@@ -394,14 +386,32 @@ func (s *Sender) muxHandleFrame(fr frames.FrameBody) error {
 			return nil
 		}
 
+		// peer is in mode second, so we must send confirmation of disposition.
+		// NOTE: the ack must be sent through the session so it can close out
+		// the in-flight disposition.
 		resp := &frames.PerformDisposition{
 			Role:    encoding.RoleSender,
 			First:   fr.First,
 			Last:    fr.Last,
 			Settled: true,
 		}
 		debug.Log(1, "TX (sender): %s", resp)
-		_ = s.l.session.txFrame(resp, nil)
+		// Ensure the session mux is not blocked
+		for {
+			select {
+			case s.l.session.tx <- resp:
+				return nil
+			case fr := <-s.l.rx:
+				// TODO: potentially recursive call
```

Comment:
Trying to decide if this potential recursion is ok (I think so but still pondering).

Reply:
Recursion has been removed.